src/app.rs

```diff
@@ -2539,12 +2539,21 @@ impl UiExtra for egui::Ui {
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-pub fn start(data_sources: Vec<Box<dyn DeferredDataSource>>) {
+pub fn start(mut data_sources: Vec<Box<dyn DeferredDataSource>>) {
     env_logger::try_init().unwrap_or(()); // Log to stderr (if you run with `RUST_LOG=debug`).
 
+    let app_name = if data_sources.len() == 1 {
+        match data_sources[0].fetch_description().source_locator {
+            Some(source_locator) => format!("{source_locator} - Legion Prof"),
+            _ => String::from("Unknown Data Source - Legion Prof"),
```

Comment:
For posterity, I'm going to mention that there are a few ways to spell this:

```rust
String::from("asdf")
"asdf".into() // requires type checker to figure out you want String
"asdf".clone() // cloning a reference gives you an owned object
"asdf".to_owned() // same, generic method for any reference to a clonable type
"asdf".to_string() // same but specific to String
```

I'm not great about consistency, but I'm trying to make a point of spelling it `"asdf".to_string()`. There's no deep reason for this, but seems to be the most common option I see in community code.